packages/safe-tools-client/app/helpers/truncate-middle.ts

```diff
@@ -0,0 +1,23 @@
+import { helper } from '@ember/component/helper';
+
+interface Signature {
+  Args: {
+    Positional: [string, number, number] | [string, number] | [string];
+  };
+  Return: string;
+}
+
+export function truncateMiddle(
+  [input, startLength = 6, endLength = 4]: [string, number, number] /*, hash*/
+) {
+  if (input && input.length <= startLength + endLength) {
+    return input;
+  } else if (input) {
+    return `${input.substring(0, startLength)}...${input.substring(
+      input.length - endLength
+    )}`;
+  }
+  return '';
+}
+
+export default helper<Signature>(truncateMiddle);
```

Comment:
I think it is time to extract an ember-helpers addon package to let us share this rather than copy/pasting it

Reply:
👍🏻 makes sense, will do, thanks

Reply:
I opened 4768 to track this.